packages/openid4vp/src/vp-token/z-vp-token.ts

```diff
@@ -1,4 +1,23 @@
 import { z } from 'zod'
 
-export const zVpToken = z.union([z.string(), z.array(z.union([z.string(), z.record(z.any())])), z.record(z.any())])
+const zVpTokenPexEntry = z.union([z.string(), z.record(z.any())], {
+  message: 'pex vp_token entry must be a string or object',
+})
+
+export const zVpTokenPex = z.union(
+  [zVpTokenPexEntry, z.array(zVpTokenPexEntry).nonempty('Must have at least entry in vp_token array')],
+  {
+    message: 'pex vp_token must be a string, object or array of strings and objects',
+  }
+)
+export type VpTokenPex = z.infer<typeof zVpTokenPex>
+export type VpTokenPexEntry = z.infer<typeof zVpTokenPexEntry>
+
+export const zVpTokenDcql = z.record(z.union([z.string(), z.record(z.any())]), {
+  message:
+    'dcql vp_token must be an object with keys referencing the dcql credential query id, and values the encoded (string or object) presentation',
```

Comment:
Is there any chance to get support for upcoming draft 25?
The main change for DCQL is that there can be multiple presentations for a credential query id, e.g.
```
{
  "credential_query_id_1": [....]
}
```

See https://github.com/openid/OpenID4VP/commit/4a6aaa9273cd120bfaed9a154a3af8397da549ef

Reply:
Yeah should be quite simple to support. We'd need to make the needed changes in the DCQL library as well.

Let's wait for 25 to be published, and then we can add support for it